dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/remote/handler/ExperimentHandler.java

```diff
@@ -0,0 +1,175 @@
+import com.dotcms.experiments.business.ExperimentsAPI;
+import com.dotcms.experiments.model.Experiment;
+import com.dotcms.publisher.pusher.wrapper.ContentTypeWrapper;
+import com.dotcms.publisher.pusher.wrapper.ExperimentWrapper;
+import com.dotcms.publisher.receiver.handler.IHandler;
+import com.dotcms.publishing.BundlerUtil;
+import com.dotcms.publishing.DotPublishingException;
+import com.dotcms.publishing.PublisherConfig;
+import com.dotcms.publishing.PublisherConfig.Operation;
+import com.dotcms.repackage.com.google.common.collect.ImmutableList;
+import com.dotcms.workflow.helper.SystemActionMappingsHandlerMerger;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.workflows.model.SystemActionWorkflowActionMapping;
+import com.dotmarketing.portlets.workflows.model.WorkflowScheme;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PushPublishLogger;
+import com.dotmarketing.util.PushPublishLogger.PushPublishAction;
+import com.dotmarketing.util.PushPublishLogger.PushPublishHandler;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import com.liferay.util.FileUtil;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * This handler class is part of the Push Publishing mechanism that deals with Experiments-related information inside a
+ * bundle and saves it in the receiving instance. This class will read and process only the {@link Experiment} data
+ * files.
+ *
+ */
+public class ExperimentHandler implements IHandler {
+
+	private final PublisherConfig config;
+	private final ExperimentsAPI experimentsAPI;
+	public ExperimentHandler(final PublisherConfig config) {
+
+		this.config = config;
+		this.experimentsAPI = APILocator.getExperimentsAPI();
+	}
+
+	@Override
+	public String getName() {
+		return this.getClass().getName();
+	}
+
+	@WrapInTransaction
+	@Override
+	public void handle(final File bundleFolder) throws Exception {
+
+	    if(LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level) {
+	        throw new RuntimeException("need an enterprise pro license to run this");
+        }
+
+		final Collection<File> experiments = FileUtil.listFilesRecursively
+				(bundleFolder, new ExperimentBundler().getFileFilter());
+
+        handleExperiments(experiments);
+	}
+
+	private void handleExperiments(final Collection<File> experiments) throws DotPublishingException {
+
+	    if(LicenseUtil.getLevel() < LicenseLevel.PROFESSIONAL.level) {
```

Comment:
double check on private method probably not needed

Reply:
Fixed